GeneralsMD/Code/GameEngine/Source/GameNetwork/GameInfo.cpp

```diff
@@ -989,13 +1021,41 @@ AsciiString GameInfoToAsciiString( const GameInfo *game )
 	}
 	optionsString.concat(';');
 
-	DEBUG_ASSERTCRASH(!TheLAN || (optionsString.getLength() < m_lanMaxOptionsLength),
-		("WARNING: options string is longer than expected!  Length is %d, but max is %d!\n",
-		optionsString.getLength(), m_lanMaxOptionsLength));
-	
 	return optionsString;
 }
 
+AsciiString GameInfoToAsciiString(const GameInfo* game)
+{
+	if (!game)
+	{
+		return AsciiString::TheEmptyString;
+	}
+
+	AsciiStringVec playerNames(MAX_SLOTS);
+	BuildPlayerNames(playerNames, *game);
+	AsciiString infoString = GameInfoToAsciiString(game, playerNames);
+
+	// TheSuperHackers @bugfix Safely truncate the game info string by
+	// scrapping characters off of player names if the overall length is too large.
+	if (infoString.getLength() > m_lanMaxOptionsLength)
```

Comment:
Perhaps we can do `TheLAN && (infoString.getLength() < m_lanMaxOptionsLength)`, judging by the DEBUG_ASSERTCRASH below.